lib/rules/require-padding-newlines-in-files.js

```diff
@@ -0,0 +1,53 @@
+ /**
+ * Requires the file to start and/or end with a newline.
+ *
+ * Types: `Boolean` or `String`
+ *
+ * Values: `true`, `"beginningOfFileOnly"` or `"endOfFileOnly"`
+ *
+ * #### Example
+ *
+ * ```js
+ * "requirePaddingNewlinesInFiles": true
+ * "requirePaddingNewlinesInFiles": "beginningOfFileOnly"
+ * "requirePaddingNewlinesInFiles": "endOfFileOnly"
+ * ```
+ */
+
+var assert = require('assert');
+
+module.exports = function() {};
+
+module.exports.prototype = {
+    configure: function(options) {
+        assert(
+            options === true || options === 'beginningOfFileOnly' || options === 'endOfFileOnly',
+            this.getOptionName() + ' option requires value to be `true`, `beginningOfFileOnly` or `endOfFileOnly`'
+        );
+
+        if (options === 'beginningOfFileOnly') {
+            this._atBOF = true;
+            this._atEOF = false;
+        } else if (options === 'endOfFileOnly') {
+            this._atBOF = false;
+            this._atEOF = true;
+        } else {
+            this._atBOF = true;
+            this._atEOF = true;
+        }
+    },
+
+    getOptionName: function() {
+        return 'requirePaddingNewlinesInFiles';
+    },
+
+    check: function(file, errors) {
+        if (this._atBOF && file.getLines()[0] !== '') {
+            errors.add('Missing padding new line at beginning of file', 1, 0);
+        }
+
+        if (this._atEOF && file.getLines()[file.getLines().length - 2] !== '') {
```

Comment:
Yeap, thanks! :+1: 

Not sure about 3. Just thinking about:
- empty file (one line) — both rules will error
- two lines:
  ```
  // ← this line will be checked by both rules
  // ← this line by nothing, should it be empty?
  ```
- three lines
  ```
  // ← this line by BOF
  // ← this line by EOF
  // should be empty?
  ```

So 3+ lines is working env for this rule at my pov. That's why I said it ;-)

Not sure, btw, what's with the last line? Should it be empty?